src/edit/selection.js

```diff
@@ -7,6 +7,10 @@ import {posFromDOM, pathToDOM, DOMFromPos, coordsAtPos} from "./dompos"
 // ;; Error type used to signal selection-related problems.
 export class SelectionError extends ProseMirrorError {}
 
+// ; Track the state of the DOM selection, firing a `selectionChange`
+// event when a selection change has occurred in a preferred manner.
+// This serves as an alternative to the native browser `selectionchange`
+// event, which is an experimental API.
```

Comment:
Since ProseMirror selections don't correspond directly to DOM selection (they can be node selections, for one thing), I don't think this characterization really holds. Having a working `selectionchange` event would remove the need to poll, but the indirection and updating the DOM to reflect our view of the selection would still be necessary.